src/lib/y2packager/known_repositories.rb

```diff
@@ -0,0 +1,98 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2019 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "yaml"
+require "yast"
+
+module Y2Packager
+  # Track the known repositories from which the system packages (drivers)
+  # have been installed (or suggested to the user).
+  # @see https://github.com/yast/yast-packager/wiki/Selecting-the-Driver-Packages
+  class KnownRepositories
+    include Yast::Logger
+
+    STATUS_FILE = "/var/lib/YaST2/system_packages_repos.yaml".freeze
+
+    # Constructor
+    def initialize
+      Yast.import "Pkg"
+      Yast.import "Installation"
+    end
+
+    def repositories
+      @repositories ||= read_repositories
+    end
+
+    def write
+      log.info("Writing known repositories #{repositories.inspect} to #{status_file}")
+
+      # accessible only for the root user, the repository URLs should not contain
+      # any passwords but rather be safe than sorry
+      File.open(status_file, "w", 0o600) do |f|
+        f.write(repositories.to_yaml)
+      end
```

Comment:
you can write it also as `File.write(status_file, repositories.to_yaml, open_args: ["w", 0o600])`

Reply:
Ah, nice trick!